ui/src/app/shared/workflow-actions.ts

```diff
@@ -0,0 +1,64 @@
+import {NodePhase, Workflow} from '../../models';
+import {services} from './services';
+import {Utils} from './utils';
+
+export type ActionDisabled = {
+    [action in WorkflowActionName]: boolean;
+};
+
+export type WorkflowActionName = 'retry' | 'resubmit' | 'suspend' | 'resume' | 'stop' | 'terminate' | 'delete';
+
+export interface WorkflowAction {
+    title: string;
+    action: () => Promise<any>;
```

Comment:
What does this `Promise` return? Does it have to be `any`?

Reply:
I made this `any` because of the `WorkflowDeleteResponse` type used here in `workflows-service.ts`: 

```js
public delete(name: string, namespace: string): Promise<WorkflowDeleteResponse> {...
```

For some reason, the delete function is the only one with its own Response type. I'll look into making a generic `WorkflowActionResponse` type to get rid of that `any`

Reply:
I see. If we don't use the result of the promise anywhere then this should be fine